consensus/src/chained_bft/block_storage/block_tree.rs

```diff
@@ -34,26 +33,16 @@ use types::crypto_proxies::LedgerInfoWithSignatures;
 /// should only be used internally in BlockStore.
 pub struct BlockTree<T> {
     /// All the blocks known to this replica (with parent links)
-    id_to_block: HashMap<HashValue, Arc<Block<T>>>,
-    /// All child links (i.e. reverse parent links) for easy cleaning.  Note that a block may
-    /// have multiple child links.  There should be id_to_blocks.len() - 1 total
-    /// id_to_child entries.
-    id_to_child: HashMap<HashValue, Vec<Arc<Block<T>>>>,
-    /// Keeps the state compute results of the executed blocks.
-    /// The state compute results is calculated for all the pending blocks prior to insertion to
-    /// the tree (the initial root node might not have it, because it's been already
-    /// committed). The execution results are not persisted: they're recalculated again for the
-    /// pending blocks upon restart.
-    id_to_compute_result: HashMap<HashValue, Arc<StateComputeResult>>,
+    id_to_block: HashMap<HashValue, ExecutedBlock<T>>,
     /// Root of the tree.
-    root: Arc<Block<T>>,
-    /// A certified block with highest round
-    highest_certified_block: Arc<Block<T>>,
+    root_id: HashValue,
```

Comment:
seems to me replace Arc with id adds one more indirection, what's your reasoning behind? avoid having Block in the tree?

Reply:
I don't have a strong opinion here with `root_id` or `root_block`.